records_mover/records/hints/conversions.py

```diff
@@ -0,0 +1,80 @@
+from .types import HintCompression, HintEncoding, HintDateFormat, HintTimeOnlyFormat, HintQuoting
+import logging
+import csv
+from typing import Optional, Dict, Union
+from typing_extensions import Literal
+
+
+logger = logging.getLogger(__name__)
+
+
+pandas_compression_from_hint: Dict[HintCompression, Optional[str]] = {
+    'GZIP': 'gzip',
+    'BZIP': 'bz2',
+    None: None,
+}
+
+
+pandas_quoting_from_hint: Dict[HintQuoting, int] = {
+    'minimal': csv.QUOTE_MINIMAL,
+    'all': csv.QUOTE_ALL,
+    'nonnumeric': csv.QUOTE_NONNUMERIC,
+    None: csv.QUOTE_NONE
+}
+
+python_encoding_from_hint: Dict[Optional[HintEncoding], str] = {
+    # valid python names: https://docs.python.org/3/library/codecs.html#standard-encodings
+    # valid hints names: https://github.com/bluelabsio/knowledge/blob/master/Engineering/
+    #    Architecture/JobDataExchange/output-design.md
+    'UTF8': 'utf-8',
+    'UTF16': 'utf-16',
+    'UTF16LE': 'utf-16-le',
+    'UTF16BE': 'utf-16-be',
+    'LATIN1': 'latin_1',
+    'CP1252': 'cp1252',
+    'UTF8BOM': 'utf-8-sig',
+    # Python will auto-detect UTF-16 with a BOM, but not UTF-8
+    'UTF16BOM': 'utf-16',
+}
+
+# TODO: this shouldn't have to be a union - what's going on?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker